src/components/icon-button/icon-button.component.js

```diff
@@ -37,6 +37,8 @@ IconButton.propTypes = {
   }).isRequired,
   /** Callback */
   onAction: PropTypes.func.isRequired,
+  /** add margin to component */
+  margin: PropTypes.string,
```

Comment:
https://github.com/Sage/carbon/blob/c615538d5ed2b812dc81fb4227e07b54d36eeb57/src/style/utils/filter-styled-system-margin-props.js

I think we should use this to define the margin prop types